octree_web_viewer/src/state.rs

```diff
@@ -0,0 +1,101 @@
+use crate::backend_error::PointsViewerError;
+use point_viewer::octree;
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+/// path information for the octrees
+#[derive(Clone)]
+pub struct OctreeKeyParams {
+    /// Location prefix, including
+    prefix: String,
+    /// Tree ID
+    suffix: String,
+}
+
+impl OctreeKeyParams {
+    pub fn get_octree_address(&self, octree_key: &String) -> Result<String, PointsViewerError> {
+        let mut join_prefix = "";
+        let mut join_suffix = "";
+
+        if !self.prefix.ends_with("/") {
+            join_prefix = "/";
+        }
+        if !self.suffix.starts_with("/") {
+            join_suffix = "/";
+        }
+
+        Ok(format!(
+            "{}{}{}{}{}",
+            self.prefix, join_prefix, octree_key, join_suffix, self.suffix
+        ))
+    }
+}
+
+#[derive(Clone)]
+pub struct AppState {
+    /// LRU Cache for Octrees
+    pub octree_map: Arc<RwLock<HashMap<String, Arc<octree::Octree>>>>,
+    /// information for retieving octree path
+    pub key_params: OctreeKeyParams,
+    /// backward compatibility to input arguments
+    pub init_uuid: String,
+}
+
+impl AppState {
+    pub fn new(
+        map_size: usize,
+        prefix: impl Into<String>,
+        suffix: impl Into<String>,
+        uuid: impl Into<String>,
+    ) -> Self {
+        AppState {
+            octree_map: Arc::new(RwLock::new(HashMap::with_capacity(map_size))),
+            key_params: OctreeKeyParams {
+                prefix: prefix.into(),
+                suffix: suffix.into(),
+            },
+            init_uuid: uuid.into(),
+        }
+    }
+
+    pub fn load_octree(
+        &self,
+        uuid: impl AsRef<str>,
+    ) -> Result<Arc<octree::Octree>, PointsViewerError> {
+        // exists
+        let octree_id = uuid.as_ref();
+        // taking care of initial octree
+        if octree_id.len() == 9 && octree_id.starts_with("init_uuid") {
```

Comment:
As mentioned above, can we pass the real id instead and get rid of this code here?

Reply:
I'd like to make this change in a new pr when possible

Reply:
done!